tenpy/algorithms/mps_sweeps.py

```diff
@@ -0,0 +1,679 @@
+"""'Sweep' algorithm and effective Hamiltonians.
+
+Many MPS-based algorithms use a 'sweep' structure, wherein local updates are
+performed on the MPS tensors sequentially, first from left to right, then from
+right to left. This procedure is common to DMRG, TDVP, sequential time evolution,
+etc.
+
+Another common feature of these algorithms is the use of an effective local
+Hamiltonian to perform the local updates. The most prominent example of this is
+probably DMRG, where the local MPS object is optimized with respect to the rest
+of the MPS-MPO-MPS network, the latter forming the effective Hamiltonian.
+
+the :class:`Sweep` class attempts to generalize as many aspects of 'sweeping'
+algorithms as possible. :class:`EffectiveH` and its subclasses implement the
+effective Hamiltonians mentioned above. Currently, effective Hamiltonians for
+1-site and 2-site optimization are implemented.
+
+.. todo ::
+- Rebuild TDVP engine as subclasses of sweep
+- Do testing
+"""
+# Copyright 2018 TeNPy Developers
+
+import numpy as np
+import time
+import warnings
+
+from ..linalg import np_conserved as npc
+from ..networks.mps import MPSEnvironment
+from ..networks.mpo import MPOEnvironment
+from ..linalg.sparse import NpcLinearOperator
+from ..tools.params import get_parameter, unused_parameters
+
+
+__all__ = ['Sweep', 'EffectiveH', 'OneSiteH', 'TwoSiteH']
+
+class Sweep:
+    """Prototype class for a 'sweeping' algorithm.
+
+    This is a superclass, intended to cover common procedures in all algorithms that 'sweep'. This
+    includes DMRG, TDVP, TEBD, etc. Only DMRG is currently implemented in this way.
+
+    .. todo ::
+    Possibly include something like plot_update_stats(), plot_sweep_stats().
```

Comment:
You can just move the `plot_*` functions from the old DMRG `Engine` into the new `DMRGEngine`, and it should work right away, I think.